### PR TITLE
Set target input to dropdown for business rules sections

### PR DESCRIPTION
**Pull request by rkrusselenq**
## Description
When adding/editing a business rule and the target type is "subsection" we need to display a drop down with each subsection.  Selecting subsections will in turn select all the questions within that subsection.

Also fixes some behavior related to setting the rule description, and some code cleanup.

## Tickets

* [Jira Ticket](https://cdc-nbs.atlassian.net/jira/software/c/projects/CNFT2/boards/99?assignee=712020%3A591f5469-427d-49e9-87dc-8b4df72a02b4&selectedIssue=CNFT2-2148)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const AddBusinessRule = () => {
                 ruleId: Number(ruleId)
             }).then((resp: Rule) => {
                 const sourceQuestion = resp.sourceQuestion?.label || '';
-
+                console.log('resp', resp);
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx:37]: Issue: console

**Review by alaapbharadwaj — CHANGES_REQUESTED** [review]

**Comment by alaapbharadwaj** [comment]
Question: Will this implementation also take care of not showing empty subsections within the dropdown? Can empty subsections be picked in classic?

**Comment by rkrusselenq** [comment]
> Question: Will this implementation also take care of not showing empty subsections within the dropdown? Can empty subsections be picked in classic?

good question, hadn't thought of that, i'll see if i can filter them out.

**Comment by rkrusselenq** [comment]
fixed

File: apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/BusinessRulesForm.tsx [hunk 2]
```diff
@@ -169,13 +182,22 @@ const BusinessRulesForm = ({ question, sourceValues }: Props) => {
 
     const handleResetSourceQuestion = () => {
         setSelectedSource([]);
-        setSourceDescription('');
+        // setSourceDescription('');
```
Comment (alaapbharadwaj) [inline apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/BusinessRulesForm.tsx:185]: Issue: comments

**Review by alaapbharadwaj — CHANGES_REQUESTED** [review]

**Comment by mpeels** [comment]
Merge conflict

**Review by alaapbharadwaj — APPROVED** [review]

**Review by mpeels — APPROVED** [review]

**Pull request merged**